crates/turborepo-api-client/src/lib.rs

```diff
@@ -325,16 +271,57 @@ impl Client for APIClient {
         }
     }
 
-    #[tracing::instrument(skip_all)]
-    async fn fetch_artifact(
+    fn make_url(&self, endpoint: &str) -> Result<Url> {
+        let url = format!("{}{}", self.base_url, endpoint);
+        Url::parse(&url).map_err(|err| Error::InvalidUrl { url, err })
+    }
+}
+
+#[async_trait]
+impl ArtifactClient for APIClient {
```

Comment:
Could we call this `CacheClient`? `Artifact` feels like an implementation detail